In [1]:
import pandas as pd
import random
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
from scipy import stats

df=pd.read_csv('IPL_Auction_2022_FullList.csv')
df

Set No. Set Name               Player       Country State Association  \
0          1        M          Trent Boult   New Zealand               NaN   
1          1        M          Pat Cummins     Australia               NaN   
2          1        M       Shikhar Dhawan         India              DDCA   
3          1        M         Shreyas Iyer         India               MCA   
4          1        M        Kagiso Rabada  South Africa               NaN   
..       ...      ...                  ...           ...               ...   
584       62    UAL15       Safyaan Sharif      Scotland               NaN   
585       62    UAL15        Henry Shipley   New Zealand               NaN   
586       62    UAL15  Maxwell Swaminathan         India               MCA   
587       62    UAL15        Johan Van Dyk  South Africa               NaN   
588       62    UAL15     Dunith Wellalage     Sri Lanka               NaN   

     Age   Specialism Batting                 Bowling    IPL  \
0     32       BOWLER     RHB    LEFT ARM Fast Medium   62.0   
1     28  ALL-ROUNDER     RHB          RIGHT ARM Fast   37.0   
2     36      BATSMAN     LHB                       -  192.0   
3     27      BATSMAN     RHB      RIGHT ARM Leg Spin   87.0   
4     26       BOWLER     LHB          RIGHT ARM Fast   50.0   
..   ...          ...     ...                     ...    ...   
584   30  ALL-ROUNDER     RHB   RIGHT ARM Fast Medium    NaN   
585   25  ALL-ROUNDER     RHB   RIGHT ARM Fast Medium    NaN   
586   23  ALL-ROUNDER     RHB      RIGHT ARM Off Spin    NaN   
587   28  ALL-ROUNDER     RHB   RIGHT ARM Fast Medium    NaN   
588   19  ALL-ROUNDER     LHB  LEFT ARM Slow Orthodox    NaN   

    Previous IPLTeam(s) 2021 Team      C/U/A  Base Price Sold Price  \
0       SRH, KKR, DD,MI        MI     Capped         200       8 CR   
1           DD, MI, KKR       KKR     Capped         200    7.25 CR   
2      DCH, MI, SRH, DC        DC     Capped         200    8.25 CR   
3                    DC        DC     Capped         200   12.25 CR   
4                    DC        DC     Capped         200    9.25 CR   
..                  ...       ...        ...         ...        ...   
584                 NaN       NaN  Associate          20        NaN   
585                 NaN       NaN   Uncapped          20        NaN   
586                 NaN       NaN   Uncapped          20        NaN   
587                 NaN       NaN   Uncapped          20        NaN   
588                 NaN       NaN   Uncapped          20        NaN   

             New Franchise     Bid  
0         Rajasthan Royals    Sold  
1    Kolkata Knight Riders    Sold  
2             Punjab Kings    Sold  
3    Kolkata Knight Riders    Sold  
4             Punjab Kings    Sold  
..                     ...     ...  
584                    NaN  UnSold  
585                    NaN  UnSold  
586                    NaN  UnSold  
587                    NaN  UnSold  
588                    NaN  UnSold  

[589 rows x 17 columns]

In [2]:
# Contingency table from ”Specialism” and ”Batting” type of player.
coc=pd.crosstab(df["Batting"], df["Specialism"])

In [3]:
coc

Specialism  ALL-ROUNDER  BATSMAN  BOWLER  WICKETKEEPER
Batting                                               
LHB                  75       36      50            13
RHB                 152       66     152            45

 From the pivot table Total LHB are: 75+36+50+13=174

 From the pivot table Total RHB are: 152+66+152+45=415

In [4]:
box = [1] * 174
box.extend([0] * 415)
random.shuffle(box)


In [5]:
def chi2(observed, expected):
    pearson_residuals = []
    for row, expect in zip(observed, expected):
        tmp=[]
        for observe in row:
            R=(observe - expect) ** 2 / expect
            tmp.append(R)
        pearson_residuals.append(tmp)
    X=np.sum(pearson_residuals)
    print(X)
    return X

In [6]:
expected_lhb = 174/4
expected_rhb = 500 - expected_lhb
expected = [expected_lhb, expected_rhb]
chi2observed = chi2(coc.values, expected)
print(f'Observed chi2: {chi2observed:.4f}')


1157.6602082310435
Observed chi2: 1157.6602


In [7]:
chisq, pvalue, df, expected = stats.chi2_contingency(coc)
print(f'Observed chi2: {chi2observed:.4f}')
print(f'p-value: {pvalue:.4f}')

Observed chi2: 1157.6602
p-value: 0.0859


Null hypothesis cannot be rejected because the p value is greater than 0.05
